add disk monitor cron

noodlemoodle opened the pull request.

ajeetgill committed:
DOS-598 DOS-653 PT-Fixes GroupBySum was breaking on DAOs. And labelize for reabable names for end user.

ajeetgill committed:
removed redundant label for count

abdelaziz-mahdy committed:
fix: improve null handling in Max and Min classes for better value comparison

noodlemoodle committed:
set pm:false daos

noodlemoodle committed:
pm true by default

sarthak-marwaha committed:
Add CIDR and enabled to System Notification Task

sarthak-marwaha committed:
move javaCode

sarthak-marwaha committed:
Add flow, move remote IP fetching

sarthak-marwaha committed:
Skip CIDR checks if no remote IP

sarthak-marwaha committed:
Small css fixes

sarthak-marwaha committed:
Update address toSummary for unstructured address

kgrgreer committed:
Improved SUM label.

sarthak-marwaha committed:
update address join to prevent stray commas

kgrgreer committed:
Don't generate java code for templates with javaSupport set to false.

sarthak-marwaha committed:
Fix edge case

jlhughes committed:
CSSAuditTest - report hard coded color and font values

jlhughes committed:
CSSAuditTest - report path relative to project.home

abdelaziz-mahdy committed:
feat: add loading spinner and animation to FlowableTree

jlhughes committed:
SecurityAuditTest - add as a first class test to be immediately aware of misconfiguration

kgrgreer committed:
Remove preventDefault and stopImmediatePropagation on FLOW Block.onClick because it was preventing links from working inside of blocks.

abdelaziz-mahdy committed:
feat: add generatedRowLabel property to GroupBy and update GROUP_BY method

kgrgreer committed:
Spacing.

kgrgreer committed:
Add UploadView which displays simpler View when uploading. Don't share dao's between upload previews.

abdelaziz-mahdy committed:
fix: correct spelling in UploadView documentation and enhance upload logic to conditionally hide controls

abdelaziz-mahdy committed:
fix: change PivotDAOAgent to extend AbstractSinkDAOAgent

abdelaziz-mahdy committed:
fix: update ID assignment logic in UploadView for matchedRows

kgrgreer committed:
Add accumulator to Pivot.toString() so that when it is used as a cache key you don't get cached values for other accumulators.

kgrgreer committed:
Start of Download DAOAgent.

kgrgreer committed:
Remove DAOPrompt.visible property which is now redundant because this is handled by the Block.

kgrgreer committed:
Make JS FObject.pub() slightly faster when there are no listeners.

kgrgreer committed:
Spacing.

kgrgreer committed:
Spacing.

kgrgreer committed:
PT-FIXES: DOS-755 Add Download DAOAgent.

kgrgreer committed:
Spacing.

kgrgreer committed:
Implement ContainsIC.toMQL().

kgrgreer committed:
Add column selection support to Download DAOAgent.

jlhughes committed:
In response to SecurityAuditTest - remove authorize(false) or add ReadOnly or add Authorizer or AuthNotes

jlhughes committed:
Crunch lite ruler - replace capablePayloadDAO with capabilityPayloadDAO as capablePayloadDAO cspec does not exists

jlhughes committed:
pom - correct typo in flags

jlhughes committed:
capablePayloadDAO - revert change, found the dao setup

jlhughes committed:
SecurityAuditTest - remove reference to DIG, not used in final version

jlhughes committed:
i18n - add GlobalReadAuthorizer

lchanmann committed:
Remove inherited regionId java validation for GBAddress

lchanmann committed:
Fix city validation

kgrgreer committed:
Spacing.

kgrgreer committed:
Spacing.

Nauna committed:
add loading property and spinner to DAOPromptView

jlhughes committed:
Jetty - enable virtual threads (if supported in JVM)

jlhughes committed:
VirtualthreadAgency - replace virtual-thread with vt for thread name to shorten log output

jlhughes committed:
jetty - 11.0.23 -> 11.0.26

jlhughes committed:
Switch default agency to virtual threadse

jlhughes committed:
Jetty virtual threads - default to false until further performance tests can be run

jlhughes committed:
HttpServer - inverted virtual thread test logic commited.

jlhughes committed:
tests - UnauthorizedRequest test required an await

jlhughes committed:
tests - CachingDAOTest test required an await

jlhughes committed:
tests - PlotTest test required an await

jlhughes committed:
JS test UnauthorizedAPIRequest fails because PasswordExpiryAuthService assumes identifier is non null, generating an NPE and return 404 rather than 401 for an invalid login attempt.

jlhughes committed:
Test - for JS tests store error output in notes so it can be reviewed later. Otherwise output is lost when tests are batch run

jlhughes committed:
Test - modelCode - skip js logic if runTest null which is the case for server side tests run from the GUI

jlhughes committed:
tests - cleanup tests copied to repo journal with 'status' values

jlhughes committed:
test - revert test cleanup - commited to wrong PR

jlhughes committed:
Test - revert use of notes. No longer needed with build support to leave headless running

abdelaziz-mahdy committed:
fix: remove default value for maintainAspectRatio property

abdelaziz-mahdy committed:
feat: add toolTip property to Command class

abdelaziz-mahdy committed:
fix: correct typo in ActionReferences comment and include toolTip in button properties

abdelaziz-mahdy committed:
fix: remove toolTip from Command properties and update ActionView to use tooltip$ binding

jlhughes committed:
Test - revert/remove notes from default table columns

sarthak-marwaha committed:
Use default foam spinner instead of a custom spinner

jlhughes committed:
TestBorder - on completion of runScript, dao reput required to await else UI passed/failed not updated

jlhughes committed:
CSSAuditTest - do not ignore important and currentColor

jlhughes committed:
Jetty11VirtualThreads - revert default threadpool - this should be in another PR - when ready

jlhughes committed:
build - default to gnuopt double-dash when no dash or double-dash proceeds a task or option, to support the very common behaviour of forgetting the double-dashses

noodlemoodle committed:
add disk monitor cron

jlhughes commented on src/foam/core/alarming/DiskSpaceMonitoringCron.js:
```diff
@@ -0,0 +1,79 @@
+  properties: [
+    {
+      class: 'Double',
+      name: 'thresholdPercent',
```
These properties can default here, but should be pulled from ScriptParameter.

jlhughes commented on src/foam/core/alarming/DiskSpaceMonitoringCron.js:
```diff
@@ -0,0 +1,79 @@
+          double percentUsed = ((total - free) / total) * 100.0;
+          if ( percentUsed > getThresholdPercent() ) {
+            note = String.format("Disk usage at %.2f%% exceeds threshold of %.2f%% on %s", percentUsed, getThresholdPercent(), getPath());
+            Alarm alarm = new Alarm.Builder(x)
```
An EventRecord should be used, which generates an alarm and also generates other notifications for real operations integration.

jlhughes commented on src/foam/core/alarming/DiskSpaceMonitoringCron.js:
```diff
@@ -0,0 +1,79 @@
+          note = String.format("Disk space usage check passed. Percentage used: %.2f%%, Alarm threshold: %.2f%%", percentUsed, getThresholdPercent());
+          Loggers.logger(x, this).info(note);
+        } catch (IOException e) {
+          note = String.format("Unable to check disk space usage for path: %s", getPath());
```
The script should throw an exception. It will be caught and reported via eventrecords.

jlhughes commented on src/foam/core/alarming/cronjobs.jrl:
```diff
@@ -112,3 +112,25 @@ p({
+  description: "Hourly cron for monitoring disk space",
+  code:
+  """
+    cron = new foam.core.alarming.DiskSpaceMonitoringCron.Builder(x).build();
```
This is just an agent that happens to executed repeatedly by the cron system.

kgrgreer committed:
Remove redundant job.complete() in SyncAssemblyLine.

kgrgreer committed:
Replace notify() with notifyAll() in AbstractAssembly.

kgrgreer committed:
Move Date Validation to own file so that it can only be included for Java. Fixes CoreTypesValidationTest.

jlhughes committed:
IntervalScheduled - refactor nextScheduledTime to start calculation at 'from' date.

Also, roll date on zero duration.
Previouly a while loop incremented the current time by the duration
until it exceeded 'from'.  When duration was zero - zero hour, minute,
second, the increment did nothing and while loop never exited.

noodlemoodle commented:
moved to https://github.com/kgrgreer/foam3/pull/4333 because I messed up the branch
